src/core/Object3D.js

```diff
@@ -662,6 +668,20 @@ Object.assign( Object3D.prototype, EventDispatcher.prototype, {
 
 		}
 
+		// SkinnedMesh specific
+
+		if ( this.skeleton !== undefined ) {
```

Comment:
Yeah, your code looks good, and this is definitely the way to go rather than piling it all into ```Object3D```.  Maybe at some point we can move the ```serialize``` and ```extractFromCache``` local convenience functions out to be either methods of the ```Object3D``` class or part of a utility class, to make it easier to do this for all cases.